pkg/endpoint/endpoint.go

```diff
@@ -153,6 +168,79 @@ type Endpoint struct {
 	PolicyMap        *policymap.PolicyMap  `json:"-"`
 	Opts             *option.BoolOptions   `json:"options"` // Endpoint bpf options.
 	Status           *EndpointStatus       `json:"status,omitempty"`
+	State            string
```

Comment:
Because Status is an array for now and state is meant to be a single state. An endpoint can only be in a single state. We can merge both together into a single type and rename status to LastStateTransitions or something but the current status is not suitable for state.